plugins/view/src/index.ts

```diff
@@ -435,9 +435,23 @@ export type ViewActionInput = 'focus' | 'selection' | 'any' | 'none'
 /**
  * @public
  */
-export type ViewAction<T = Record<string, any>> = Resource<
-(doc: Doc | Doc[] | undefined, evt: Event, params?: T) => Promise<void>
->
+export type ViewActionFunction<TDoc extends Doc = Doc, TParams = Record<string, any>> = (
+  doc: TDoc | TDoc[] | undefined,
+  evt: Event,
+  params?: TParams
+) => Promise<void>
+
+/**
+ * @public
+ */
+export type ViewActionAvailabilityFunction<TDoc extends Doc = Doc> = (
```

Comment:
Let's use not a TDoc, since it conflict with our model definition classes and could confuse somebody.

Reply:
`<TSmth extends Smth>` is kinda standard for generics in TS when you need longer names, but I see how it clashes with our convention for class types, so let's come up with an alternative convention. How about `<GSmth extends Smth>` (_G_ standing for _generic_)?

Reply:
Nevermind, in this exact place we don't even need longer names

Reply:
Fixed